hazelcast/src/main/java/com/hazelcast/security/permission/JobPermission.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.security.permission;
+
+public class JobPermission extends InstancePermission {
+
+    private static final int READ = 4;
+    private static final int LIST = 8;
+    private static final int SUSPEND = 16;
+    private static final int RESUME = 32;
+    private static final int EXPORT = 64;
+    private static final int UPLOAD = 128;
+    private static final int ALL = CREATE | DESTROY | READ | LIST | SUSPEND | RESUME | EXPORT | UPLOAD;
+
+
+    public JobPermission(String... actions) {
+        super("<job>", actions);
+    }
+
+    @Override
+    protected int initMask(String[] actions) {
+        int mask = READ;
```

Comment:
Doesn't this mean that `READ` will be available, even if `actions` is empty? Every other implementation of this method starts with `NONE`.

Reply:
yes, this means read will be granted if you have any other permission. it does not makes sense to have submit permission but not read permission.